gnc/sub8_controller/nodes/pd_controller.cc

```diff
@@ -0,0 +1,218 @@
+/*
+  This is a simple PD Controller designed for a few reasons:
+    - Prove the validity of the existing simulator
+      - And to serve as a tool for setting simulated waypoints without directly setting position
+    - Prove the controller unit-test / monte-carlo system as a useful tool for testing
+    - Experiment with C++ Dynamic Reconfigure
+    - Start filling out the gnc/control stuff
+
+  It is NOT:
+    - Intended to be a serious replacement for the sub7 controller
+        (Something else is cooking to replace that)
+*/
+#include "sub8_controller/pd_controller.h"
+
+PDController::PDController() {
+  reconfigure_callback = boost::bind(&PDController::gain_callback, this, _1, _2);
+  server.setCallback(reconfigure_callback);
+  // Wrench Publisher
+  wrench_pub = nh.advertise<geometry_msgs::WrenchStamped>("wrench", 1);
+  // Target Waypoint
+  waypoint_sub = nh.subscribe("trajectory", 1, &PDController::waypoint_callback, this);
```

Comment:
Why did you call this callback "waypoint_callback"? Instead of "trajectory_callback"?